data/jakarta/test-security-services-proxy-certs.sh

```diff
@@ -124,45 +126,49 @@ else
     echo "Self-signed TLS verification recheck test succeeded"
 fi
 
+# generate CA-signed TLS certificate using openssl
 
+# Generate the Certificate Authority (CA) Private Key
+openssl ecparam -name prime256v1 -genkey -noout -out ca.key
+# Generate the Certificate Authority Certificate
+openssl req -new -x509 -sha256 -key ca.key -out ca.crt -subj "/CN=checkbox-test-ca"
+# Generate the Server Certificate Private Key
+openssl ecparam -name prime256v1 -genkey -noout -out server.key
+# Generate the Server Certificate Signing Request
+openssl req -new -sha256 -key server.key -out server.csr -subj "/CN=localhost"
+# Generate the Server Certificate
+openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -days 1000 -sha256
 
-echo "Installing edgeca if missing"
-if [ -z "$(snap list edgeca)" ]; then
-    snap install edgeca
-    edgeca_is_installed=true
-    echo "edgeca installed"
-fi
+TEST_CERT=$(< server.crt)
+TEST_KEY=$(< server.key)
 
-# wait for edgeca service to come online
-snap_wait_port_status 50025 open
-
-# generate CA-signed TLS certificate
-# We are running the test script with 'sudo' and although the edgeca snap has the home interface, 
-# which allows access to the home directory, when running as sudo, the user is root, 
-# so it has a different home directory and doesn't have write access to your home directory. 
-# It's therefore easiest to use the $SNAP_DATA directory of the EdgeCA snap:
-EDGECA_DIR="/var/snap/edgeca/current"
-echo "Generating CSR"
-edgeca gencsr --cn localhost --csr $EDGECA_DIR/csrfile --key $EDGECA_DIR/csrkeyfile
-echo "Generating certificate"
-edgeca gencert -o $EDGECA_DIR/localhost.cert -i $EDGECA_DIR/csrfile -k $EDGECA_DIR/localhost.key
-EDGECA_CERT=$(< $EDGECA_DIR/localhost.cert)
-EDGECA_KEY=$(< $EDGECA_DIR/localhost.key)
-
-echo "Setting security-proxy certificate "
-snap set edgexfoundry env.security-proxy.tls-certificate="$EDGECA_CERT"
+echo "Setting security-proxy certificate"
+snap set edgexfoundry env.security-proxy.tls-certificate="$TEST_CERT"
 echo "Setting security-proxy certificate private key"
-snap set edgexfoundry env.security-proxy.tls-private-key="$EDGECA_KEY" 
+snap set edgexfoundry env.security-proxy.tls-private-key="$TEST_KEY"
 
 # the CA certificate needs to be where edgexfoundry.curl can read it
 echo "Copying CA certificate"
-cp $EDGECA_DIR/CA.pem $EDGEXFOUNDRY_SNAP_DATA/
+cp ca.crt $EDGEXFOUNDRY_SNAP_DATA/
+
+echo "Checking that Kong is using new certificate"
+until [[ $(edgexfoundry.curl --silent http://localhost:8001/certificates | edgexfoundry.jq -r ".data[0].key") == $TEST_KEY ]]; do
+    sleep 1
+    iter_num=$(( iter_num + 1 ))
+    echo "Checking Kong TLS certificate..."
+    if [ $iter_num -gt 10 ]; then
+        print_error_logs
+        >&2 echo "Failed to set Kong TLS certificate"
+        snap_remove
+        exit 1
+    fi
+done
```

Comment:
Please move to data/utils.sh and call the function from jakarta and latest suites.
Also, please print the counter info to make it look like:
Checking Kong TLS certificate... 1/10

Reply:
fixed